WordPress/gutenberg
# #39457 image block: keep image size on changing image

## What?
This PR fixes #39457 to keep the selected size in an image block when replacing/changing the image.


## Why?

Useful for patterns where a certain image size is required for it to look good. A pattern can declare the sizeSlug attribute and it will try to insert the new image in the size again.

## How?

Previously on selecting a new image there was a check if the new image is available in the default image size, if yes it used that and if not it falled back to "full". Now it checks for a previously selected sizeSlug attribute and checks if the image is available in that size, if not it does the same checks as before (default size, full). So the order is:
1. previously selected image size (from block attribute)
2. default image size
3. "full" image size

## Testing Instructions




1. Insert an image block in the editor
2. pick an image
3. image should be inserted in the default image size ("large")
4. select any other size than the default size
5. click "replace" and choose another image
6. the image should be inserted in the previously selected size and the sizeSlug dropdown in the sidebar should also contain this size

1. Create a pattern which includes an image block with a predefined size
2. insert the pattern
3. set/replace the image with any other image
5. the image should be inserted in the previously selected size and the sizeSlug dropdown in the sidebar should also contain this size

## Screenshots or screencast
